src/torrent_handle.cpp

```diff
@@ -691,6 +691,11 @@ namespace libtorrent {
 		return sync_call_ret<storage_interface*>(nullptr, &torrent::get_storage_impl);
 	}
 
+    void torrent_handle::set_sequential_start(piece_index_t piece) const
+    {
+        async_call(&torrent::set_sequential_start, piece);
```

Comment:
It would seem more appropriate to set the start and end at the same time as sequential mode is enabled. It really should be just one round-trip to the network thread too.

What do you think of a function that enables sequential mode and takes a start- and end piece as well?

Reply:
No. You don't understand.
The start piece must be set asynchronously at any point in time.
And the sooner it applies, then the better.
The start can change regularly while the data is being downloaded.
So combining this with the sequential mode enable function is not possible.

Specifying the end piece is not necessary.

Reply:
setting it at the same time as you set sequential mode is as soon as it can set, so, the best, right?

Reply:
specifying end is necessary if you don't want sequential mode past a certain point

Reply:
> setting it at the same time as you set sequential mode is as soon as it can set, so, the best, right?

No. The start of data downloading in sequential mode can start as usual. There is no need to set 0. The starting piece will need to be set some time after the start of downloading.

> specifying end is necessary if you don't want sequential mode past a certain point

There is no such scenario. This function is definitely not needed.

Reply:
> No. The start of data downloading in sequential mode can start as usual.

That's suboptimal. Why create an API that invites suboptimal behavior?

> There is no such scenario. This function is definitely not needed.

How about, I want to download file 2 sequentially, the other files I don't care about. (i.e. they should be rarest first)

Reply:
@arvidn 

> setting it at the same time as you set sequential mode is as soon as it can set, so, the best, right?

Okay. I looked at my code. Indeed, sometimes there is a need to specify the start before the start of loading.

Let's add this function in the next pull request.